src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -214,12 +235,17 @@ public NetworkPeerConnection(NetworkPeer peer, NetworkPeerClient client, IDateTi
                 {
                     this.logger.LogTrace("Exception occurred: '{0}'", ex.ToString());
 
-                    this.Peer.State = NetworkPeerState.Failed;
-                    this.Peer.DisconnectReason = new NetworkPeerDisconnectReason()
+                    if (this.Peer.DisconnectReason == null)
                     {
-                        Reason = "Unexpected exception while sending a message",
-                        Exception = ex
-                    };
+                        this.Peer.DisconnectReason = new NetworkPeerDisconnectReason()
+                        {
+                            Reason = "Unexpected exception while sending a message",
```

Comment:
Add '.' here.

Reply:
I have removed dots from these reasons because of how these reasons are displayed to the console. The original code also lack those dots.